superset-frontend/src/explore/components/DatasourcePanel.tsx

```diff
@@ -159,36 +160,65 @@ const LabelContainer = styled.div`
   }
 `;
 
-const DataSourcePanel = ({
+export default function DataSourcePanel({
   datasource,
   controls: { datasource: datasourceControl },
   actions,
-}: Props) => {
+}: Props) {
   const { columns, metrics } = datasource;
   const [lists, setList] = useState({
     columns,
     metrics,
   });
-  const search = ({ target: { value } }: { target: { value: string } }) => {
+
+  const search = debounce((value: string) => {
     if (value === '') {
       setList({ columns, metrics });
       return;
     }
     setList({
       columns: matchSorter(columns, value, {
-        keys: ['column_name', 'expression', 'description', 'verbose_name'],
+        keys: [
+          'column_name',
+          'verbose_name',
+          {
+            key: 'description',
+            threshold: rankings.STARTS_WITH,
```

Comment:
Isn't this going to eliminate all the columns which description does not start with a search keyword and put all the columns that start with search keyword on top of the list?

Reply:
![Screenshot 2021-01-22 at 17 58 44](https://user-images.githubusercontent.com/26679866/105521118-c81eff80-5cdb-11eb-91ca-ea01169b0566.png)
![Screenshot 2021-01-22 at 17 58 37](https://user-images.githubusercontent.com/26679866/105521122-c9502c80-5cdb-11eb-85ae-f08a8edfcc11.png)

Reply:
notice that **calc_first_time_dev** is missing from the list

Reply:
It was a bug. This should use `CONTAINS` like in `metrics`. Thanks for catching this!